### DISCUSSION TITLE
Is it possible to downgrade the version of GlobaLeaks?

### DISCUSSION THREAD

**Discussion by elbill** [globaleaks/globaleaks-whistleblowing-software#4556] [Q&A] [answered]
**Describe the bug**
In the scenario that one has an older GL db eg 4.9.9 and wants to migrate it to a new system.
I install current version of GL (4.10.7) and works fine (accessible trough http (IP address) and tor
I downgrade to 4.9.9  apt-get install globaleaks=4.9.9
However after this the platform is not accessible through tor nor http
When updated to 4.10.7  works again

**Expected behavior**
I would expect the platform to be accessible after downgrading
Ubuntu 20.04 pro LTS
4.10.7
Downgraded to 4.9.9

**Comment by evilaliv3** [comment, marked as answer]
Dear @elbill 

The scenario described is not supported by the application because the software do not support any downgrade.

Let me clarify:
When you update from one version to a newer version the database is updated and would typically not work anymore with a previous release of the software. If you do so you will incurr in similar issues like the one that you are running into.
If you really need to downgrade, you need the original directory /var/globaleaks archived before updating and corresponding to the same database format of the older globaleaks version.
To restore a previous version, restore a copy of your old data in the position /var/globaleaks and install the corresponding version.

**Comment by elbill** [comment]
I'm not using a new database with an old version. Trying to create an old version to feed it the old database. In my scenario thta I have 4.9.9 db, how can I create a system on another vm to be able to access it?

In a few words how can I create a 4.9.9 GL system today?

**Comment by elbill** [comment]
I'm trying to make a functional system 4.9.9 by installing a 4.10.7 and then using apt-get install globaleaks=4.9.9
If I uncompress the 4.9.9 db in /var/globaleaks will that work?

**Comment by evilaliv3** [comment]
The current install script wont work for this use case. I will work editing it in order to add a parameter to specify a software version.

**Comment by evilaliv3** [comment]
@elbill: I've modified the installation script does now accept a parameter -v where you can indicate a software version

you can use it like: ```./install.sh -v 4.9.9```

**Comment by elbill** [comment]
Thats great. Thanks!
